packages/flutter/lib/src/widgets/router.dart

```diff
@@ -0,0 +1,1228 @@
+    scope.routerState._setStateWithExplicitReportStatus(
+      _RouteInformationReportingStatus.must,
+      callback
+    );
+  }
+
+  /// Forces the [Router] to to run the [callback] without reporting the route
+  /// information back to the engine.
+  ///
+  /// Use this method if you don't want the [Router] to report the new route
+  /// information even if it detects changes as a result of running the
+  /// [callback].
+  ///
+  /// See also:
+  ///
+  ///  * [Router]: see the "URL updates for web applications" section for more
+  ///    information about route information reporting.
+  static void neglect(BuildContext context, VoidCallback callback) {
+    final _RouterScope scope = context.getElementForInheritedWidgetOfExactType<_RouterScope>().widget as _RouterScope;
+    scope.routerState._setStateWithExplicitReportStatus(
+      _RouteInformationReportingStatus.ignore,
+      callback
+    );
+  }
+
+  @override
+  StatefulElement createElement() => _RouterStatefulElement(this);
+
+  @override
+  State<Router<T>> createState() => _RouterState<T>();
+}
+
+typedef _AsyncPassthrough<Q> = Future<Q> Function(Q);
+
+class _RouterStatefulElement extends StatefulElement {
+  _RouterStatefulElement(StatefulWidget widget) : super(widget);
+
+  _RouterState<dynamic> get routerState => state as _RouterState<dynamic>;
+
+  @override
+  void rebuild() {
+    super.rebuild();
+    routerState._reportRouteInformation();
+  }
+}
+
+// Whether to report the route information in this build cycle.
+enum _RouteInformationReportingStatus {
+  // We haven't receive any signal on whether to report.
+  none,
+  // Report if route information changes.
+  maybe,
+  // Report regardless of route information changes.
+  must,
+  // Don't report regardless of route information changes.
+  ignore,
+}
+
+class _RouterState<T> extends State<Router<T>> {
+  Object _currentRouteInformationParserTransaction;
+  Object _currentRouterDelegateTransaction;
+  _RouteInformationReportingStatus _reportingStatus;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.routeInformationProvider?.addListener(_handleRouteInformationProviderNotification);
+    widget?.backButtonDispatcher?.addCallback(_handleBackButtonDispatcherNotification);
+    widget.routerDelegate.addListener(_handleRouterDelegateNotification);
+    _reportingStatus = _RouteInformationReportingStatus.none;
+    if (widget.routeInformationProvider != null) {
+      _processInitialRoute();
+    }
+  }
+
+  void _reportRouteInformation() {
+    switch(_reportingStatus) {
+      case _RouteInformationReportingStatus.none:
+      case _RouteInformationReportingStatus.ignore:
+        _reportingStatus = _RouteInformationReportingStatus.none;
+        return;
+      case _RouteInformationReportingStatus.maybe:
+      case _RouteInformationReportingStatus.must:
+        final T configuration = widget.routerDelegate.currentConfiguration;
```

Comment:
I would have expected that we compare this configuration to whatever the last reported configuration was and when it has changed, we run it through `restoreRouteInformation` and `routerReportsNewRouteInformation`?